### DISCUSSION TITLE
Can not connect to Wireguard VPN server using EthernetWebServer Library

### DISCUSSION THREAD

**Discussion by MahdiBabaie** [khoih-prog/EthernetWebServer#80] [Q&A]
Hello to all
Hardware: ESP 32 + W 5500 Module
Software: PlatformIO + Visual Studio Code
Libraries used: WireGuard-ESP32.h + EthernetWebServer.h
You can see the above libraries in the following links on github.
https://github.com/ciniml/WireGuard-ESP32-Arduino
https://github.com/khoih-prog/EthernetWebServer
I can connect to Wireguard VPN Server using WiFi connection and WireGuard-ESP32.h library and everything works fine and sending and receiving data is fine.
When using the EthernetWebServer.h library, the device is connected to the Internet and can be seen and pinged in the Local network, and even when connected to the Wireguard VPN Server, the function wg.is_initialized() returns the value 1, but in sending and receiving data There is a problem.
Where do you think the problems are?

**Comment by khoih-prog** [comment]
Try to use ESP32 WiFi-related `LwIP` libraries, with the ESP32 built-in SSL feature

1. [**WebServer_ESP32_W5500**](https://github.com/khoih-prog/WebServer_ESP32_W5500)
2. [**AsyncWebServer_ESP32_W5500**](https://github.com/khoih-prog/AsyncWebServer_ESP32_W5500)

instead of this library.

Certainly, if still not OK, only you can solve the issue for your special `WireGuard` use-case, especially without [**MRE**](https://stackoverflow.com/help/minimal-reproducible-example)

Good Luck,

**Reply by MahdiBabaie** [reply]
I did not understand your meaning
If possible, explain in more detail.
Are you saying that I should use the WebServer_ESP32_W5500 or AsyncWebServer_ESP32_W5500 library instead of the EthernetWebServer library?
And one more question: What do you mean that I have to do this without MRE?
